apps/prairielearn/src/pages/instructorEffectiveUser/instructorEffectiveUser.html.ts

```diff
@@ -0,0 +1,345 @@
+import { format, toZonedTime } from 'date-fns-tz';
+import { z } from 'zod';
+
+import { html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+import {
+  CourseInstancePermissionSchema,
+  CoursePermissionSchema,
+  UserSchema,
+} from '../../lib/db-types.js';
+
+export const CourseRolesSchema = z.object({
+  available_course_roles: CoursePermissionSchema.shape.course_role.unwrap().array(),
+  available_course_instance_roles: CourseInstancePermissionSchema.shape.course_instance_role
+    .unwrap()
+    .array(),
+  available_uids: UserSchema.shape.uid.array().nullable(),
+});
+export type CourseRoles = z.infer<typeof CourseRolesSchema>;
+
+export function InstructorEffectiveUser({
+  resLocals,
+  ipAddress,
+  courseRoles,
+}: {
+  resLocals: Record<string, any>;
+  ipAddress: string;
+  courseRoles: CourseRoles;
+}) {
+  const { authz_data, course_instance, __csrf_token, req_date, true_req_date, user } = resLocals;
+
+  // This page can be mounted under `/pl/course/...`, in which case we won't
+  // have a course instance to get a display timezone from. In that case, we'll
+  // fall back to the course, and then to the institution. All institutions must
+  // have a display timezone, so we're always guaranteed to have one.
+  const displayTimezone =
+    resLocals.course_instance?.display_timezone ??
+    resLocals.course.display_timezone ??
+    resLocals.institution.display_timezone;
+
+  const formattedTrueReqDate = format(
+    toZonedTime(true_req_date, displayTimezone),
+    "yyyy-MM-dd'T'HH:mm:ss.SSSxxx",
+    { timeZone: displayTimezone },
+  );
+  const formattedReqDate = format(
+    toZonedTime(req_date, displayTimezone),
+    "yyyy-MM-dd'T'HH:mm:ss.SSSxxx",
+    { timeZone: displayTimezone },
+  );
```

Comment:
I have kept the current behaviour of using `date-fns-tz`'s version of format, since it needs to be suitable for `parseISO` in the input associated to changing the date. If there is a need to change to another format (e.g., from `@prairielearn/formatter`) then this would also require a change in the description of the format below the input, as well as the parsing function.